skiko/src/jvmMain/kotlin/org/jetbrains/skiko/GraphicsApi.kt

```diff
@@ -10,4 +14,45 @@ enum class GpuPriority(val value: String) {
     companion object {
         fun parse(value: String?): GpuPriority? = GpuPriority.values().find { it.value == value }
     }
-}
\ No newline at end of file
+}
+
+private val notSupportedAdapters by lazy {
+    val resource = SkiaLayer::class.java.getResource("/not-supported-adapter.list").readText()
+    resource.split(";").map { it.trim() }
+}
+
+internal fun isVideoCardSupported(renderApi: GraphicsApi): Boolean {
+    return when (renderApi) {
+        GraphicsApi.DIRECT3D -> {
+            val adaptersList = notSupportedAdapters.filter { it.startsWith("directx:") }.map {
+                it.replace("directx:", "")
+            }
+            var adapter = getNextDirectXAdapter()
+            while (adapter != null) {
+                adaptersList.forEach {
+                    if (adapter!!.startsWith(it)) {
+                        return false
+                    }
+                }
+                adapter = getNextDirectXAdapter()
```

Comment:
If the first adapter is supported, and second isn't - we switch to Software

Maybe we should just check the first adapter?

But I believe we should use the same logic as in defineHardwareAdapter, or better not separate them into two places.

If we need to check adapter as soon as possible, we shouldn't write the same code twice - we should call createDevice as soon as possible. The dispose problem (see above) will also be fixed if we will do that.